### DISCUSSION TITLE
create custom window

### DISCUSSION THREAD

**Discussion by Stepixo** [godotengine/godot-proposals#7524] [GUI]
### Describe the project you are working on

I want to create a custom window so that I can control the top bar of the window, change its appearance and color, and add some buttons or backgrounds to it

If so, what is the way to enable the window to work as I want it? An example of the format I'm trying to reach is avast.

### Describe the problem or limitation you are having in your project

I tried to use the borderless option, but in this case I can no longer move or scale the window. Is this the normal behavior of the borderless option, that is, it prevents me from resizing or moving the window
If so, what is the way to enable the window to work as I want it? An example of the format I'm trying to reach is avast.
![jRIf2](https://github.com/godotengine/godot-proposals/assets/110250940/3b18e0d6-4839-4925-bdcf-ca5837ac2864)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

What I suggest is to add a new feature to enable the user of Godot to control the top bar of the main window of the application and change its colors and buttons either by deleting or adding or even adding a background extending from the top bar to include the window area

The other solution is to modify the way the borderless property works in project settings to remove the top bar, but maintain the ability to move and resize the window

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Add the property in project settings

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I don't know

### Is there a reason why this should be core and not an add-on in the asset library?

I think that Godot , even if it is mainly for developing games, also offers great features for developing programs, and to keep up with the modern software interfaces , so I think it is good to have this feature in the engine and this will enable developers to fully customize the interfaces of their applications

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot-proposals/issues/4846.

Moving to discussion, as there is no concrete technical implementation provided.

> Is this the normal behavior of the borderless option, that is, it prevents me from resizing or moving the window

Yes, this is expected when using a borderless window. When using a borderless window, there are no borders that you can drag to move or resize the window in the first place.

**Reply by Stepixo** [reply]
So is there a way to achieve what I want using gdscript  , What I mean is to use the borderless property and then use code to move the window and resize it. As for close ,maximize , minimize , I did that 
There is a simple problem, which is that when I minimize the window, it happens correctly, but when I click on the program icon in the Windows taskbar, the window does not reappear
